builder/googlecompute/startup.go

```diff
@@ -1,37 +1,40 @@
 package googlecompute
 
 import (
-	"encoding/base64"
 	"fmt"
 )
 
-const StartupScriptStartLog string = "Packer startup script starting."
-const StartupScriptDoneLog string = "Packer startup script done."
 const StartupScriptKey string = "startup-script"
+const StartupScriptStatusKey string = "startup-script-status"
 const StartupWrappedScriptKey string = "packer-wrapped-startup-script"
 
-// We have to encode StartupScriptDoneLog because we use it as a sentinel value to indicate
-// that the user-provided startup script is done. If we pass StartupScriptDoneLog as-is, it
-// will be printed early in the instance console log (before the startup script even runs;
-// we print out instance creation metadata which contains this wrapper script).
-var StartupScriptDoneLogBase64 string = base64.StdEncoding.EncodeToString([]byte(StartupScriptDoneLog))
+const StartupScriptStatusDone string = "done"
+const StartupScriptStatusError string = "error"
+const StartupScriptStatusNotDone string = "notdone"
 
-var StartupScript string = fmt.Sprintf(`#!/bin/bash
-echo %s
+var StartupScriptLinux string = fmt.Sprintf(`#!/bin/bash
+echo "Packer startup script starting."
 RETVAL=0
+BASEMETADATAURL=http://metadata/computeMetadata/v1/instance/
 
 GetMetadata () {
-	echo "$(curl -f -H "Metadata-Flavor: Google" http://metadata/computeMetadata/v1/instance/attributes/$1 2> /dev/null)"
+	echo "$(curl -f -H "Metadata-Flavor: Google" ${BASEMETADATAURL}/${1} 2> /dev/null)"
 }
 
-STARTUPSCRIPT=$(GetMetadata %s)
+ZONE=$(GetMetadata zone | grep -oP "[^/]*$")
+
+SetMetadata () {
+	gcloud compute instances add-metadata ${HOSTNAME} --metadata ${1}=${2} --zone ${ZONE}
```

Comment:
So this add the dependency of `gcloud` on them source image. Is that necessary? Could you use curl instead?